docs/scos/dev/upgradability-services/new-documentation/core-method-used-on-project-level.md

```diff
@@ -0,0 +1,119 @@
+---
+title: A private API is used on the project level
+description: Reference information for evaluator and upgrader tools.
+last_updated: Mar 23, 2022
+template: concept-topic-template
+---
+
+Modules have public and private APIs. While public API updates always support backward compatibility, private API updates can break backward compatibility. So, backward compatibility is not guaranteed in the private API. For example, if you use a core method on the project level, and it is updated or removed, it can cause unexpected issues during updates.
+
+For more information about module APIs, see [Definition of Module API](/docs/scos/dev/architecture/module-api/definition-of-module-api.html).
+
+## Example of code that causes the upgradability error
+
+`CustomerFacade` uses the `createCustomerReader` and `getCustomerCollection`. This is a  private API.
+
+```php
+namespace Pyz\Zed\Customer\Business;
+
+use Spryker\Zed\Customer\Business\CustomerFacade as SprykerCustomerFacade;
+
+/**
+ * @method \Pyz\Zed\Customer\Business\CustomerBusinessFactory getFactory()
+ */
+class CustomerFacade extends SprykerCustomerFacade implements CustomerFacadeInterface
+{
+   /**
+     * ...
+     */
+    public function getCustomerCollection(...): ...
+    {
+        return $this->getFactory()
+            ->createCustomerReader()
+            ->getCustomerCollection(...);
+    }
+}
+```
+
+## Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------------------------------------------
+Pyz\Zed\Customer\Business\CustomerFacade
+"PrivateApi:Persistence Please avoid Spryker dependency: "{name}""
+************************************************************************************************************************
+```
+
+## Example of resolving the Evaluator check error
+
+To resolve the error provided in the example, do the following steps:
```

Comment:
1. Do I need to check the strategies in this particular order?
2. If one of the strategies can be used to resolve an issue, should I use that strategy and discard the rest?